api/v1beta1/openstackmachine_types.go

```diff
@@ -89,6 +90,12 @@ type OpenStackMachineSpec struct {
 	// credentials specified in the cluster will be used.
 	// +optional
 	IdentityRef *OpenStackIdentityReference `json:"identityRef,omitempty"`
+
+	// FloatingAddressFromPool is a reference to a IPPool that will be assigned
+	// to an IPAddressClaim. Once the IPAddressClaim is fulfilled, the FloatingIP
+	// will be assigned to the OpenStackMachine.
+	// +optional
+	FloatingAddressFromPoolRef *corev1.TypedLocalObjectReference `json:"floatingAddressFromPool,omitempty"`
```

Comment:
I know this has been up for discussion. I changed to `FloatingIPPoolRef` now however know that @mdbooth and @bilbobrovall have discussed this one prior. Maybe you have any input on this?

Reply:
It's a reference to a pool that will be used for a single floating address associated with the machine and it's been discussed and renamed a couple of times already but navigating old discussions isn't easy on github..

But I think the original name is clearer since we might want to reference different pools for different things in the future and it's **not** a "anytime this machine wants an fip grab it from this pool"-kind of thing

Reply:
I prefer `FloatingIPPoolRef` because:
* it fits with k8s naming conventions on object references by ending in `Ref`
* it accurately describes the requirement of the pool, which must return a floating IP